### PR TITLE
add new function

### PR DESCRIPTION
**Pull request by travis-sobeck**
Add function Get-GrouperStemByParent for improved searches of Stems

### CODE REVIEW

File: UMN-Grouper.psm1 [hunk 1]
```diff
@@ -376,6 +376,95 @@ function Get-GrouperPrivileges
     }
 #endregion
 
+#region Get-GrouperStemByParent
+    function Get-GrouperStemByParent
+    {
+        <#
+            .SYNOPSIS
+                Get Grouper chled Stem(s) of a parent stem
```
Comment (FISHMANPET) [inline UMN-Grouper.psm1]:
```suggestion
                Get Grouper child Stem(s) of a parent stem
```

File: UMN-Grouper.psm1 [hunk 2]
```diff
@@ -376,6 +376,95 @@ function Get-GrouperPrivileges
...
+
+            .DESCRIPTION
+                Get Grouper chled Stem(s) of a parent stem
```
Comment (FISHMANPET) [inline UMN-Grouper.psm1]:
```suggestion
                Get Grouper child Stem(s) of a parent stem
```

**Review by FISHMANPET — COMMENTED** [review]
Wow this API's documentation really sucks...

From what I can decipher this seems reasonable.  And I'm guessing the 'chled' is a typo, apparently I can suggest changes right in the review now, so you can accept them directly into your branch.

Also, have you tested the case where you only get a single response?  I keep running into cases where powershell helpfully strips the container off of return value when there's only one value, so it no longer has a count parameter, so doing 'count -gt 0' doesn't work.  If it strips the container away you can replace that with ($null -eq [...].stemResults) or even simply if ([..]stemResults)

**Comment by FISHMANPET** [comment]
wait no I think I found something.  According to this, ALL_IN_SUBTREE is the default parentStemNameScope: https://cdn.rawgit.com/Internet2/grouper/master/grouper-ws/grouper-ws/doc/api/edu/internet2/middleware/grouper/ws/coresoap/WsStemQueryFilter.html?view=co#getParentStemNameScope()

So in theory it's recursive by default.  To be consistent with the API should the cmdlet be recursive by default, and only be non-recursive if specified.  Maybe a switch called NoRecursion or OneLevelOnly to be consistent with the terminology used in the API?

**Comment by travis-sobeck** [comment]
Added a new function initially to avoid breaking Get-GrouperStem but I'm not sure it necessary, might be better served by revamping Get-GrouperStem.

File: UMN-Grouper.psm1 (hunk 2 above)
Comment (travis-sobeck) [reply UMN-Grouper.psm1]: I'll test the case of a single a single return value.  Do you have a specific value you used so I an duplicate your test as?

**Comment by travis-sobeck** [comment]
I also need to do a little more testing around how FIND_BY_STEM_NAME_APPROXIMATE functions

**Comment by travis-sobeck** [comment]
Also, yeah I'm good with making our functions default keep in-line with the api's default of ALL_IN_SUBTREE

**Comment by travis-sobeck** [comment]
adding Get-GrouperStemByUUID (just because i was at it).
Fixed Get-GrouperStem to work more like I would expect, it gets a single stem unless you use the search param.
Also fixed Get-GrouperStemByParent to be explicit about recursion.  Although the docs say 'ALL_IN_SUBTREE' that's not how our version actually works, however I set it to follow the directions of docs in case the API changes, in any case, removed any reliance on a "default"

**Comment by FISHMANPET** [comment]
Chatted with Travis, this all looks good.  Future state could be to collapse all the get stem functions into a single function with parameter sets, or write a private function to make the actual call (since the 3 functions have a lot of identical code) but this is good.

**Review by FISHMANPET — APPROVED** [review]
Chatted with Travis a bit, these 3 functions look good.  A future state could be collapsing them into one (since they're using the same API call) or writing a private function to make the actual API call and have these be wrappers that assemble the specific data, but these will work.

**Pull request merged**